Tensile/KernelWriterAssembly.py

```diff
@@ -11124,9 +11231,15 @@ def closePrefetchAcrossPersistent(self, kernel):
   def functionEnd(self, kernel, addLabel=True):
     imod = Code.Module()
     if kernel["PersistentKernel"]:
+
+      if kernel["StoreRemapVectorWidth"]:
+        imod.addInst("s_barrier", "")
```

Comment:
Would it be better to put the barrier immediately before the local writes instructions in the next loop? So that you can always have buffer loads (for the next loop) sooner before hitting the barrier

Reply:
Yes, thanks for the review. I've changed this, now the barrier located after ShadowInit and before the first ds_write